railgun/test/integration/perf_ingress_test.go

```diff
@@ -0,0 +1,129 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
+	k8sgen "github.com/kong/kubernetes-testing-framework/pkg/generators/k8s"
+)
+
+func TestIngressPerf(t *testing.T) {
+	t.Log("setting up the TestIngressPerf")
+	proxyReady()
+
+	ctx := context.Background()
+
+	cnt := 1
+	cost := 0
+	for cnt < 1 {
+
+		namespace := fmt.Sprintf("ingress-%d", cnt)
+		nsName := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		_, err := cluster.Client().CoreV1().Namespaces().Create(context.Background(), nsName, metav1.CreateOptions{})
+		assert.NoError(t, err)
+
+		t.Logf("[%s] deploying a minimal HTTP container deployment to test Ingress routes", namespace)
+		err = configManifest(ctx, namespace, "httpbin.yaml", t)
+		assert.NoError(t, err)
+
+		t.Logf("[%s] List the service from the namespace.", namespace)
+		svcs, err := cluster.Client().CoreV1().Services(namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Logf("Failed to list services.")
+			continue
+		}
+
+		var service *v1.Service
+		for _, svc := range svcs.Items {
+			fmt.Printf("\n service %v \n", svc)
+			if svc.Name == "httpbin" {
```

Comment:
If what we are trying to accomplish is just to find this service, we can avoid a `Services(ns).List()` altogether and just use `Services(ns).Get(ctx, "httpbin")` instead